pkg/systemd/hwinfo.jsx

```diff
@@ -24,61 +24,195 @@ import React from "react";
+                        <th>{ _("Name") }</th>
+                        <td>{ info.name }</td>
                     </tr>
                     <tr>
-                        <th>{ _("BIOS date") }</th>
-                        <td>{ moment(info.bios_date).isValid() ? moment(info.bios_date).format('L') : info.bios_date }</td>
+                        <th>{ _("Version") }</th>
+                        <td>{ info.version }</td>
                     </tr>
-                </React.Fragment>
-            }
-            { info.nproc !== undefined &&
-                <tr>
-                    <th>{ _("CPU") }</th>
-                    <td>{ (info.nproc > 1) ? `${info.nproc}x ${info.cpu_model}` : info.cpu_model }</td>
-                </tr>
+                </tbody>
+                }
+                <tbody>
+                    { info.bios_vendor &&
+                    <React.Fragment>
+                        <tr>
+                            <th>{ _("BIOS") }</th>
+                            <td>{ info.bios_vendor }</td>
+                        </tr>
+                        <tr>
+                            <th>{ _("BIOS version") }</th>
+                            <td>{ info.bios_version }</td>
+                        </tr>
+                        <tr>
+                            <th>{ _("BIOS date") }</th>
+                            <td>{ moment(info.bios_date).isValid() ? moment(info.bios_date).format('L') : info.bios_date }</td>
+                        </tr>
+                    </React.Fragment>
+                    }
+                    { info.nproc !== undefined &&
+                    <React.Fragment>
+                        <tr>
+                            <th>{ _("CPU") }</th>
+                            <td>{ (info.nproc > 1) ? `${info.nproc}x ${info.cpu_model}` : info.cpu_model }</td>
+                        </tr>
+                        { onSecurityClick !== undefined &&
+                        <tr>
+                            <th>{ _("CPU Security") }</th>
+                            <td>{ (this.state.allowed && <a role="link" onClick={onSecurityClick}>{ _("Mitigations") }</a>) ||
+                                <OverlayTrigger overlay={ <Tooltip id="tip-cpu-security">{ cockpit.format(_("The user $0 is not permitted to change cpu security mitigations"), permission.user ? permission.user.name : '') }</Tooltip> }><span>{ _("Mitigations") }</span></OverlayTrigger> }
+                            </td>
+                        </tr>
+                        }
+                    </React.Fragment>
+                    }
+                </tbody>
+            </table>
+        );
+    }
+}
+
+class CPUSecurityMitigationsDialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.close = this.close.bind(this);
+        this.saveAndReboot = this.saveAndReboot.bind(this);
+        this.state = {
+            nosmt: undefined,
+            alert: undefined,
```

Comment:
This is missing an `available`. It's not technically required, but nice as a summary which fields can be in the state.